docs/csharp/how-to/compare-strings.md

```diff
@@ -0,0 +1,86 @@
+
+[!code-csharp-interactive[Searching in a sorted array](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#6)]
+
+## Case sensitive and culture sensitive sorting and searching
+
+The following code uses the <xref:System.Collections.Generic.List%601?displayProperty=nameWithType> collection class to store strings. The strings are sorted using the <xref:System.Collections.Generic.List%601.Sort%2A?displayProperty=nameWithType> method. This method needs a delegate that compares and orders two strings. The <xref:System.String.CompareTo%2A?displayProperty=nameWithType> method provides that comparison function. Run the sample and observe the order. This sort operation is both case sensitive and culture sensitive. You would use the static <xref:System.String.Compare%2A?displayProperty=nameWithType> methods to specify different comparison rules.
+
+[!code-csharp-interactive[Sorting a list of strings](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#7)]
+
+Once sorted, the list of strings can be searched using a binary search. The following sample shows how to search the sorted listed using the same comparison function. The local function `ShowWhere` shows where the sought text is or would be:
+
+[!code-csharp-interactive[csProgGuideStrings#11](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#8)]
+
+Collection classes such as <xref:System.Collections.Hashtable?displayProperty=nameWithType>, <xref:System.Collections.Generic.Dictionary%602?displayProperty=nameWithType>, and <xref:System.Collections.Generic.List%601?displayProperty=nameWithType> have constructors that take a <xref:System.StringComparer?displayProperty=nameWithType> parameter when the type of the elements or keys is `string`. In general, you should use these constructors whenever possible, and specify either <xref:System.StringComparer.Ordinal?displayProperty=nameWithType> or <xref:System.StringComparer.OrdinalIgnoreCase?displayProperty=nameWithType>.
+
+## Comparing across cultures
+
+The following example shows several aspects of comparing strings across cultures. A string is initialized to the phrase "They dance in the street." in German. Linguistically (in Windows), "ss" is equal to the German essetz: 'ß' character in both "en-US" and "de-DE" cultures. This sample stores <xref:System.Globalization.CultureInfo> for the current culture. The original culture can be set and retrieved on the current thread object. The comparisons are performed using the <xref:System.StringComparison.CurrentCulture> value to ensure a culture-specific comparison.
```

Comment:
Should *essetz* be *Eszett* (based on [wiki page](https://en.wikipedia.org/wiki/%C3%9F))?